pkg/migrations/sql/idempotent/base.sql

```diff
@@ -613,6 +613,33 @@ $$
 LANGUAGE SQL STABLE PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION SCHEMA_PROM.label_key_position(text, text) to prom_reader;
 
+-- drop_metric deletes a metric and related series hypertable from the database along with the related series, views and unreferenced labels.
+CREATE OR REPLACE FUNCTION SCHEMA_PROM.drop_metric(metric_name_to_be_dropped text) RETURNS VOID
+AS
+$$
+    DECLARE
+        hypertable_name TEXT;
+        deletable_metric_id INTEGER;
+    BEGIN
+        IF (SELECT NOT pg_try_advisory_lock(SCHEMA_LOCK_ID)) THEN
```

Comment:
Sorry this should be a transaction not session-level lock
```suggestion
        IF (SELECT NOT pg_try_advisory_xact_lock(SCHEMA_LOCK_ID)) THEN
```